File: scrapy/crawl_zhihu.go
package main

import (
	"fmt"
	"io/ioutil"
	"net/http"
)

func main() {
	url := "https://www.zhihu.com/"
	request, err := http.NewRequest("GET", url, nil)
	if err != nil {
		panic(err)
	}

	request.Header.Set("User-Agent", "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/68.0.3440.106 Safari/537.36")
	request.Header.Set("cookie", `@@@`)

	client := http.DefaultClient
	response, err := client.Do(request)
	if err != nil {
		panic(err)
	}
	fmt.Println(response.Status)

	html, err := ioutil.ReadAll(response.Body)
	if err != nil {
		panic(err)
	}
	err = ioutil.WriteFile("zhihu.html", html, 0666)
	if err != nil {
		panic(err)
	}
}

File: scrapy/crawl_zhihu.py
import os
import requests


url = 'https://www.zhihu.com/signup'
login_url = "https://zhihu-web-analytics.zhihu.com/api/v1/logs/batch"
headers = {
    "accept": "*/*",
    "accept-encoding": "gzip, deflate, br",
    "accept-language": "zh-CN,zh;q=0.9",
    "content-encoding": "gzip",
    "content-length": "465",
    "content-type": "application/x-protobuf",
    "origin": "https://www.zhihu.com",
    "referer": "https://www.zhihu.com/signup?next=%2F",
    "user-agent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko)Chrome/68.0.3440.106 Safari/537.36",
    "x-za-batch-size": "1",
    "x-za-log-version": "2.3.91",
    "x-za-platform": "1",
    "x-za-product": "1",
}


resp = requests.post(login_url, headers=headers, data=data)
# resp.raise_for_status()
print(resp.status_code)
resp.encoding == resp.apparent_encoding
print(resp.text.encode(resp.encoding).decode())
